File: 643-maximum-average-subarray-i/maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
       double maxSum;
       double sum;
       for(int i=0;i<k;i++){
        sum+=nums[i];
       }
       maxSum= sum;

        for (int i = k; i < nums.size(); i++) {
            sum += nums[i] - nums[i - k]; // Add the new element and subtract the old one
            maxSum = max(maxSum, sum);   // Update maxSum if the current window's sum is greater
        }
        return maxSum/k;
    }
};

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var findMaxAverage = function(nums, k) {
    
    let result =0;
    let average =0;
    let sum =0;
    for(let i=0;i<k;i++){
         sum += nums[i];

    }
    average = sum/k;
    result = average;


    for(let i=k;i<nums.length;i++){
        sum = sum- nums[i-k] + nums[i];
        average = sum/k;
        result = Math.max(average,result);
    }
    return result;
    
};